### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by rayman89** [gorhill/uBlock#231]
Sorry if this is not the place for this if it's not please point me to where this should go.

I'm trying to test ublock but is hard for me to migrate from adblock plus cause the interface is very different this add-on looks more like the typical chrome ad blocker add-on and i never got those so my questions are:

Is there a way to quickly port the customizations I did on adblock plus to ublock? If so how can i do that? (I'm using firefox)

What's the difference between my filters my rules and whitelist? 

How do I add personal exceptions network/cosmetic rules and what's the sintaxis for them? 

How do I disable particular rules of subscription filters? (for example in ESP nauscopic filters  "publicidad/*")

Is ublock e10s tab ready?

If your main "selling" point is that Ublock is better than others are you planing in making migration easier and or more intuitive?

**Comment by gorhill** [comment]
> Is there a way to quickly port the customizations I did on adblock plus to ublock? If so how can i do that? (I'm using firefox)

There is not much to customize, the biggest part being custom filters. In the _"My filters"_ pane, there is an _"Import and append"_ button. You can imported an ABP back up file, uBlock will extract the custom filters from it.

> What's the difference between my filters my rules and whitelist? 

_"My filters"_ is for static filters, i.e. those compatible with ABP filter syntax, or compatible with hosts file syntax.

_"My rules"_ are all rules related to [dynamic filtering](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering).

> How do I add personal exceptions network/cosmetic rules and what's the sintaxis for them?

In the _"My filters"_ pane, using ABP filter syntax. If you want to completely disable uBlock for certain sites, it's the _"Whitelist"_ pane.

> How do I disable particular rules of subscription filters? (for example in ESP nauscopic filters "publicidad/*")

You cannot disable specific filters (hence "static"). You can override them with other custom filters, i.e. using your example, `@@publicidad/*`.

> Is ublock e10s tab ready?

Yes.

> If your main "selling" point is that Ublock is better than others

My point it's that uBlock is resource-efficient and that all features are to serve users' interests and nobody else ("selling", whether in quotes or not is not part of my mindset).

I don't think migrating is an issue I need to work on, I don't see any big obstacle to this. I am not planning any specific work for this.

**Comment by rayman89** [comment]
Tanks for your quick and clear response. By "selling" point what I meant is that  that was the focus to encourage users to use this add-on over similar ones.

If you import the ABP file it has all the filter rules so you gonna get duplicates of the same filters you subscribed before so it's not as easy as you mentioned you have to dig trough the file to search for the specific filter lines.

There is also an issue on firefox when you go to options on the add-on list there are two buttons that appear at the bottom this don't say anything but the upper one takes you to the actual options and the other to devtools.html

**Comment by gorhill** [comment]
> By "selling" point what I meant is that that was the focus to encourage users to use this add-on

You will never find me encouraging users to use uBlock over something else. I publish benchmark results, I describe feature set, and if someone ask what to use, I tell them to choose themselves.

> If you import the ABP file it has all the filter rules so you gonna get duplicates of the same filters you subscribed before so it's not as easy as you mentioned

You are making assumptions. Try it and see for yourself.

> the other to devtools.html

What?

**Comment by gwarser** [comment]
> > the other to devtools.html
> 
> What?

Logger in old version.

**Comment by gorhill** [comment]
> Logger in old version

Actually, now that you mention this, this should be renamed "Show Logger", "Show Network Request Logger" is no longer accurate.

**Comment by rayman89** [comment]
Why would you publish benchmarks and compare it to others if you are not trying to say "Hey mine is better"? I don't see anything bad with what you are doing I'm just explaining myself.

http://www.imagebam.com/image/e1db40411130089

The second button takes you to chrome://ublock/content/devtools.html

I did not make assumptions I tried. Remember I'm using firefox not chrome. Firefox ABP and chrome ABP are very different. To start with when you export the ABP filters from the ABP windows you get an ini file and ublock requires a txt file. Then it has syntax like this

[Filter]
text=||syndication.twimg.com/widgets/$third-party
disabled=true

[Subscription]
url=https://easylist-downloads.adblockplus.org/easylist.txt
title=EasyList
fixedTitle=true
homepage=https://easylist.adblockplus.org/
lastDownload=1432263977
downloadStatus=synchronize_ok
lastSuccess=1432263977
lastCheck=1432313394
expires=1432955177
softExpiration=1432626532
version=201505220300
requiredVersion=2.0
downloadCount=27

[Subscription filters]
! Last modified: 22 May 2015 03:00 UTC
! Licence: https://easylist-downloads.adblockplus.org/COPYING
!
! Please report any unblocked adverts or problems
! in the forums (https://forums.lanik.us/)
! or via e-mail (easylist.subscription@gmail.com).
!
!-----------------------General advert blocking filters-----------------------!
! *** easylist:easylist/easylist_general_block.txt ***
&ad_box_
&ad_channel=
&ad_classid=
&ad_height=


and starts listing all the filters

**Comment by gorhill** [comment]
> Why would you publish benchmarks and compare it to others if you are not trying to say "Hey mine is better"?

You are ascribing motives. The alternative is that I should keep secret the benchmarks I use to drive open development. Do you think this make sense?

> I did not make assumptions I tried.

You're right. Last time I tried it it worked. The issue is when there is no custom filters. It works if there are custom filters. I will fix this.

**Comment by gorhill** [comment]
ABP backup import import fixed in 73ac79feff76ecd71d5aeb15e05fc8486d5d8e3c#diff-af2d24ded69a121983a18b6743ff3558

**Comment by rayman89** [comment]
What I'm saying is that there should be a motivation for you to post the benchmarks as a comparative else you wouldn't do it. The benchmarks you post show that uBlock better than the other options this is not a description of the program functionality so why does it exist if it's not to prove that is better than the others and therefore attract more users to use your add-on instead of the others?

**Comment by alejandrolemus** [comment]
Hi @rayman89, I collaborate with the Spanish translation. Would you mind sharing where [that screenshot](http://www.imagebam.com/image/e1db40411130089) comes from? Mine doesn't look like that in Firefox 38.0.1

**Comment by rayman89** [comment]
@alejandrolemus

I checked and it seems that the issue is that https://addons.mozilla.org/en-US/firefox/addon/ublock/ is distributing version 0.9.1.0 and that's the version that I had on version 0.9.3.5 the buttons are displayed correctly.

**Comment by alejandrolemus** [comment]
@rayman89 Thanks! 
So you were (are?) using uBlock, ([chris' version](https://github.com/chrisaljoudi/uBlock)), not uBlock Origin (gorhill's version, this repo).
Not sure if that makes a difference regarding your OP.

**Comment by gorhill** [comment]
> The benchmarks you post show that uBlock better than the other options this is not a description of the program functionality

Efficiency is also a feature.

> there should be a motivation

So what was my motivation in informing people that uBlock did not have all the features ABP had back in early development?

https://github.com/gorhill/uBlock/wiki/What-%C2%B5Block-can-and-can-not-(currently)-do/c38e0b11918e8661438ad212b74dd2a485482cba#%C2%B5block-has-all-the-features-abp-has

Anyway, I was merely trying to answer your questions, make what you want of my answers. I am not a fan of endless back and forth in Github issues, you can have the last word.

**Comment by alejandrolemus** [comment]
@gorhill 
FWIW, I'm pretty sure he didn't mean anything funny. It seems English it's not his first language, hence the wording choice (been a Spanish speaker myself, I can easily understand why 'selling point' was used).

**Comment by rayman89** [comment]
@alejandrolemus 
I was using the original now I'm using origin.

@gorhill 
Sorry for the non-technical discussion.

BTW if you search for ublock origin on the mozilla webpage it doesn't show up for me. I had to use the link provided here. I dont know if this is an issue with mozilla webpage or something.
